packages/synthetix-main/contracts/storage/Distribution.sol

```diff
@@ -238,16 +232,16 @@ library Distribution {
      */
     function getActorValueChange(Data storage dist, bytes32 actorId) internal view returns (int valueChange) {
         DistributionActor.Data storage actor = dist.actorInfo[actorId];
-        int128 deltaValuePerShare = dist.valuePerShare - actor.lastValuePerShare;
+        int128 deltaValuePerShare = dist.valuePerShareD27 - actor.lastValuePerShareD27;
 
-        int changedValueHighPrecision = deltaValuePerShare * actor.shares.uint128toInt256();
-        valueChange = changedValueHighPrecision.fromHighPrecisionDecimalToInteger();
+        int changedValueHighPrecision = deltaValuePerShare * actor.sharesD18.uint128toInt256();
```

Comment:
mixing D27 and D18 here. Check if maths work

Reply:
Yep, it goes to D45, then to D18 on the next line